runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3269,6 +3275,11 @@ remoteCompile(
                   } // end switch (it->getKind())
                }
             metaData->runtimeAssumptionList = *(compiler->getMetadataAssumptionList());
+
+            for (auto& it : methodsRequiringTrampolines)
+               {
+               compInfoPT->reloRuntime()->codeCache()->reserveResolvedTrampoline(it, true);
```

Comment:
The generate code and the trampoline reservations need to stay in the same code cache. If we cannot reserve a trampoline here we need to take some corrective action, possibly leading to failing the compilation.
if you look at how we reserve trampolines in the non-jitserver case you'll see it's quite complicated.

Reply:
I see that in `TR_J9VMBase::getResolvedTrampoline` failing to allocate a trampoline always results in failing the compilation when we are in binary encoding. If I understand correctly, it's because we start populating the code cache in the binary encoding phase, so we can't just start using a new cache at that point and have to fail a compilation.
I think for JITServer it would be reasonable to fail with `TR::RecoverableTrampolineError` if `reserveResolvedTrampoline` returns an error code because by that point we already received the code cache from the server so we can't just create a new one.

Reply:
ok